### PR TITLE
PENG-183 - Add get-vault-secrets workflow template

### PR DESCRIPTION
**Pull request by WillBartee**
### Related to
- https://github.com/myfitnesspal/github-actions/pull/15
- https://github.com/myfitnesspal/mfp-terraform-atlantis/pull/76
- https://github.com/myfitnesspal/infra-docker/pull/16

### CODE REVIEW

File: workflow-templates/get-vault-secrets.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        ref: main
+        ssh-key: ${{ secrets.MFP_GITHUB_ACTIONS_SSH_KEY }}
+        path: .github/common-actions
+# ------------------------------------------------------------
+    - name: Get Secrets and AWS Creds
+      id: generic
+      uses: ./.github/common-actions/get-vault-secrets
+      with:
+        secretId: ${{ secrets.vault_secret_id }}
+        exportAWS: true   # Required for AWS credentials. NOTE: Make sure you have requested access through mfp-terraform-atlantis
```
Comment (CKarper) [inline workflow-templates/get-vault-secrets.yml:20]: #AnswerPlease: Are we providing a default AWS role for all builds?  Or should this be `false` by default unless overridden?

File: workflow-templates/get-vault-secrets.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        secrets:
+          # TODO Fill this out for secrets stored in Vault at path "ci/<your-repo>/some-login"
```
Comment (CKarper) [inline workflow-templates/get-vault-secrets.yml:22]: #AnswerPlease: Are we able to just inject all secrets from the repo's vault path by default?

**Review by CKarper — APPROVED** [review]
I know this is a Nit, but other actions have an icon...  Can we add a vault SVG to this PR and make it the action icon?

File: workflow-templates/get-vault-secrets.yml (hunk 2 above)
Comment (WillBartee) [reply workflow-templates/get-vault-secrets.yml:22]: This is a possibility, but not implemented.

File: workflow-templates/get-vault-secrets.yml (hunk 1 above)
Comment (WillBartee) [reply workflow-templates/get-vault-secrets.yml:20]: Q: _Are we providing a default AWS role for all builds?_
A: No. We could add one, but it shouldn't have any permissions.

Q: _should (`exportAWS`) be `false` by default unless overridden?_
A: It is `false` by default.

File: workflow-templates/get-vault-secrets.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        repository: myfitnesspal/github-actions
+        ref: main
+        ssh-key: ${{ secrets.MFP_GITHUB_ACTIONS_SSH_KEY }}
+        path: .github/common-actions
+# ------------------------------------------------------------
+    - name: Get Secrets and AWS Creds
+      id: generic
+      uses: ./.github/common-actions/get-vault-secrets
+      with:
+        secretId: ${{ secrets.vault_secret_id }}
```
[the remark this answers is not in the document]
Comment (WillBartee) [reply workflow-templates/get-vault-secrets.yml:19]: GitHub lists secrets in the UI in all caps, however the code to fetch secrets is case insensitive. That said, I'm happy to make that change to be consistent.

File: workflow-templates/get-vault-secrets.yml (hunk 2 above)
Comment (gaffneyd4) [reply workflow-templates/get-vault-secrets.yml:22]: It feels fine to have consumers of this action pick which secrets they want to use. They less in GitHub actions land the better 🤷

**Review by gaffneyd4 — APPROVED** [review]

**Pull request merged**